File: main/c++/algorithms/comb_sort.cpp
#include <cmath>

#include "sort.h"

int *comb_sort (int *array, int len) {
    int gap = len;
    bool swapped = true;
    while (gap !=1 || swapped == false) {
        gap = floor((gap * 10)/13);
        if (gap < 1) {
            gap = 1;
        }

        swapped = false;
        for (int i = 0; i < len-gap; i++) {
            if (array[i] > array[i + gap]) {
                int temp = array[i];
                array[i] = array[i + gap];
                array[i + gap] = temp;
                swapped = true;
            }
        }
    }
    return array;
}

File: main/python/algorithms/comb_sort.py
def comb_sort(array):
    n = len(array)
    gap = n
    swapped = True
    while gap !=1 or swapped == False:
        gap = (gap * 10)//13
        if gap < 1: gap = 1

        swapped = False
        for i in range(0, n-gap):
            if array[i] > array[i + gap]:
                temp = array[i]
                array[i] = array[i + gap]
                array[i + gap] = temp
                swapped = True
    
    return array